drivers/modem/hl7800.c

```diff
@@ -4627,7 +4626,7 @@ static struct net_offload offload_funcs = {
 
 static inline uint8_t *hl7800_get_mac(struct device *dev)
 {
-	struct hl7800_iface_ctx *ctx = dev->driver_data;
+	struct hl7800_iface_ctx *ctx = dev->data;
```

Comment:
The struct device changes should be in a separate commit from the newlibc changes. Both need a better commit message explanation.